cli/src/pcluster/commands.py

```diff
@@ -352,138 +351,87 @@ def _colorize(stack_status, args):
-        sys.exit(1)
+        result = PclusterApi().list_clusters(region=utils.get_region())
+        if isinstance(result, list):
+            clusters = [[cluster.name, _colorize(cluster.status, args), cluster.version] for cluster in result]
+            LOGGER.info(tabulate(clusters, tablefmt="plain"))
+        else:
+            utils.error(f"Unable to retrieve the list of clusters.\n{result.message}")
     except KeyboardInterrupt:
         LOGGER.info("Exiting...")
         sys.exit(0)
 
 
-def _poll_head_node_state(stack_name):
-    ec2 = boto3.client("ec2")
-    try:
-        instances = utils.describe_cluster_instances(stack_name, node_type=utils.NodeType.head_node)
-        if not instances:
-            LOGGER.error("Cannot retrieve head node status. Exiting...")
-            sys.exit(1)
-        head_node_id = instances[0].get("InstanceId")
-        state = instances[0].get("State").get("Name")
-        sys.stdout.write("\rMasterServer: %s" % state.upper())
-        sys.stdout.flush()
-        while state not in ["running", "stopped", "terminated", "shutting-down"]:
-            time.sleep(5)
-            state = (
-                ec2.describe_instance_status(InstanceIds=[head_node_id])
-                .get("InstanceStatuses")[0]
-                .get("InstanceState")
-                .get("Name")
-            )
-            head_node_status = "\r\033[KMasterServer: %s" % state.upper()
-            sys.stdout.write(head_node_status)
-            sys.stdout.flush()
-        if state in ["terminated", "shutting-down"]:
-            LOGGER.info("State: %s is irrecoverable. Cluster needs to be re-created.", state)
-            sys.exit(1)
-        head_node_status = "\rMasterServer: %s\n" % state.upper()
-        sys.stdout.write(head_node_status)
-        sys.stdout.flush()
-    except ClientError as e:
-        LOGGER.critical(e.response.get("Error").get("Message"))
-        sys.stdout.flush()
-        sys.exit(1)
-    except KeyboardInterrupt:
-        LOGGER.info("\nExiting...")
-        sys.exit(0)
-
-    return state
-
-
-def _get_compute_instances(stack):
-    instances = utils.describe_cluster_instances(stack, node_type=utils.NodeType.compute)
-    return map(lambda i: ("ComputeFleet", i.get("InstanceId")), instances)
-
-
 def instances(args):
-    stack_name = utils.get_stack_name(args.cluster_name)
-    PclusterConfig.init_aws(config_file=args.config_file)
-    cfn_stack = utils.get_stack(stack_name)
-    scheduler = utils.get_cfn_param(cfn_stack.get("Parameters"), "Scheduler")
-
-    instances = []
-    head_node_server = utils.describe_cluster_instances(stack_name, node_type=utils.NodeType.head_node)
-    if head_node_server:
-        instances.append(("MasterServer", head_node_server[0].get("InstanceId")))
+    """Print the list of instances associated to the cluster."""
+    try:
+        result = PclusterApi().describe_cluster(cluster_name=args.cluster_name, region=utils.get_region())
```

Comment:
Why use utils.get_region()? Do we have --region arg for this command?

Reply:
In the main of the cli.py there is:
```python
        # set region in the environment to make it available to all the boto3 calls
        if "region" in args and args.region:
            os.environ["AWS_DEFAULT_REGION"] = args.region
```

and utils.get_region() does:
```python
    return os.environ.get("AWS_DEFAULT_REGION") or boto3.session.Session().region_name
```

So get_region is safer than args.region, because args.region can be empty but boto3 region must be populated.
The cli must pass a region to the "api" in any case, even if args.region is not passed.

The idea is:
- if cli main found the region sets in the env
- starting from there all the code can use get_region.